lib/WP/Notice.php

```diff
@@ -12,37 +12,41 @@ class Notice {
   private $message;
   private $classes;
   private $data_notice_name;
+  private $render_in_paragraph;
 
-  function __construct($type, $message, $classes = '', $data_notice_name = '') {
+  function __construct($type, $message, $classes = '', $data_notice_name = '', $render_in_paragraph = true) {
```

Comment:
This is getting ridiculous. We should refactor this class instead of adding another optional argument. First of all, it is hard to understand what each does without looking at the code. Second of all do you have to use this class? cannot you display the notice from frontend?

Reply:
I agree, that it's getting ridiculous, but at the same time, I don't think it should be part of this issue to refactor it (I can create Jira issue for that). It's perfectly usable for this use case, but the way it was designed in the past (always show title and render inside `<p>`, which then forbid using HTML inside error message, which is already used in code) does not work with specs of this issue, so I had to add those arguments to prevent that behavior but also accommodate new specs.

Reply:
Jira issue for refactor https://mailpoet.atlassian.net/browse/MAILPOET-1831